### ISSUE TITLE
Containerize xCAT

### ISSUE THREAD

**Issue by robin2008** [xcat2/xcat2-task-management#405]
Target to Containerize so as to:
- run xCAT easily on an existing OS without change it
- integrated with k8s or other container management(ICP) to act as a bunch of services: like PXE, HW-ctl, Remote-Console, etc.

This is a long term EPIC,  will add more tasks when the investigation go far. Below just an rough list.
- [  ] The initial docker file to build container for xCAT (Centos7,  Ubuntu16) , migrated from existing implementation. 
- [  ] Investigate how to include into auto test environment. 
- [  ] Using `supervisord` to control services, instead of `systemd`
- [  ] Integrated with DB (MariaDB, Postgresql)
- [  ] How to handle the logs - docker host must have run syslog
- [  ] Integrated with existing DHCP, to see if we can leverage `dnsmasq` to only offer the proxy service
- [  ] Possible Microservices (goconserver, hw-control, etc)

**Comment by robin2008** [comment]
Only parts of will be covered in 2.14.5 parts

**Comment by whowutwut** [comment]
@robin2008 I'd like to also make a comment that if we plan to support a Docker image in public docker hub, we need to first have a way to automatically create this docker image on each release of xCAT, i don't want this to become a maintenance issue for anyone to spend time on each release to generate and manually push a new image to the docker hub.    If we don't have this, should be one of the first things..

**Comment by robin2008** [comment]
Yes, it will be integrated it in each build/release procedure once we have done.

**Comment by zet809** [comment]
Do you think we shall have a basic container manager or scheduler such as swarm, compose to do the container management in xCAT cluster?
